src/Cassandra/SSLOptions.php

```diff
@@ -24,29 +24,83 @@
  */
 final class SSLOptions
 {
-  /**
-   * Trusted certificate paths. This is used to verify node's identity.
-   * @var array|null
-   */
-  public $trustedCerts = null;
-  /**
-   * Client cert
-   * @var string|null
-   */
-  public $clientCert = null;
-  /**
-   * Path to private key, this is used to verify client's identity.
-   * @var string|null
-   */
-  public $privateKey = null;
-  /**
-   * Optional passphrase for the private key
-   * @var string|null
-   */
-  public $passphrase = null;
-  /**
-   * A bitmask of Cassandra::VERIFY_* constants
-   * @var int|null
-   */
-  public $verifyFlags = null;
+    /**
+     * Trusted certificate paths. This is used to verify node's identity.
+     * @var array|null
+     */
+    private $trustedCerts = null;
+
+    /**
+     * Client cert
+     * @var string|null
+     */
+    private $clientCert = null;
+
+    /**
+     * Path to private key, this is used to verify client's identity.
+     * @var string|null
+     */
+    private $privateKey = null;
+
+    /**
+     * Optional passphrase for the private key
+     * @var string|null
+     */
+    private $passphrase = null;
+
+    /**
+     * A bitmask of Cassandra::VERIFY_* constants
+     * @var int|null
+     */
+    private $verifyFlags = null;
+
+    /**
+     * Constructs SSLOptions instance
+     * Don't instantiate directly, use Cassandra\SSLOptions\Builder instead.
+     *
+     * @access private
+     * @param array|null $trustedCerts Trusted certificate paths
+     * @param int|null   $verifyFlags  A bitmask of Cassandra::VERIFY_* constants
+     * @param string     $clientCert   Client cert
+     * @param string     $privateKey   Path to private key
+     * @param string     $passphrase   Passphrase for the private key if any
+     */
+    public function __construct($trustedCerts, $verifyFlags, $clientCert, $privateKey, $passphrase)
+    {
+        $this->trustedCerts = $trustedCerts;
+        $this->verifyFlags  = $verifyFlags;
+        $this->clientCert   = $clientCert;
+        $this->privateKey   = $privateKey;
+        $this->passphrase   = $passphrase;
+    }
+
+    /**
+     * Creates ssl context resource
+     * @access private
+     * @return resource ssl context
+     */
+    public function resource()
+    {
+        $resource = cassandra_ssl_new();
+
+        if (is_array($this->trustedCerts)) {
```

Comment:
what about typehinting it as array to avoid this check ? ``null`` is currently strictly equivalent to an empty array, so you could just use an empty array